src/test/java/com/juniorstart/juniorstart/service/ProjectServiceTest.java

```diff
@@ -0,0 +1,228 @@
+        projectRequest.setTitle(project.getTitle());
+    }
+
+    private void initializeTechnology() throws Exception {
+        technology = Technologies.builder().technologyType(TechnologyType.FRONTEND).description("Some description").id(1l).title("Some title").build();
+        technology.setCreatedAt(createdAt);
+        technology.setUpdatedAt(createdAt);
+        technology.setCreatedBy(user.getPublicId());
+        technology.setUpdatedBy(user.getPublicId());
+    }
+
+    @BeforeEach
+    public void initialize() throws Exception {
+
+        createdAt = new SimpleDateFormat("yyyy-MM-dd").parse("2020-12-31").toInstant();
+        users = new ArrayList<>();
+        initializeUser();
+        initializeProject();
+        projectList = new ArrayList<>();
+        projectList.add(project);
+        page = MockUtil.createMockPage(projectList);
+        userPrincipal = UserPrincipal.create(user);
+        initializeProjectRequest();;
+        initializeTechnology();
+        project.getTechnologies().add(technology);
+        project.setRepository("Some Repo");
+        project.setUpdatedAt(createdAt);
+        projectSet = new HashSet<>();
+        projectSet.add(project);
+        technology.setProjects(projectSet);
+    }
+
+    @Test
+    public void should_return_findAll_method() throws Exception {
+       when(projectRepository.findAll(isA(Pageable.class))).thenReturn(page);
+       when(userDao.findByPublicId(anyLong())).thenReturn(Optional.of(user));
+
+        PagedResponse<ProjectResponse> response = projectService.getAllProjects(0 , 10);
+        assertTrue(response.getContent().get(0).getBody().contains(project.getBody()));
+        assertTrue(response.getContent().get(0).getDescription().contains(project.getDescription()));
+        assertTrue(response.getContent().get(0).getName().contains(project.getName()));
+        assertTrue(response.getContent().get(0).getTitle().contains(project.getTitle()));
+
+        verify(projectRepository, times(1)).findAll(isA(Pageable.class));
+        verify(userDao, times(1)).findByPublicId(anyLong());
+    }
+
+
+    @Test
+    public void should_create_new_project() throws Exception {
+
+        when(userDao.findById(ArgumentMatchers.any(UUID.class))).thenReturn(Optional.of(user));
+        when(projectRepository.findByName(ArgumentMatchers.any(String.class))).thenReturn(Optional.empty());
+        when(projectRepository.save(ArgumentMatchers.any(Project.class))).thenReturn(project);
+        Project response = projectService.createProject(userPrincipal, projectRequest);
+        assertTrue(response.getBody().contains(project.getBody()));
+        assertTrue(response.getName().contains(project.getName()));
+        assertEquals(response.getRecruiting(), project.getRecruiting());
+        verify(userDao, times(1)).findById(any(UUID.class));
+        verify(projectRepository, times(1)).findByName(anyString());
+        verify(projectRepository, times(1)).save(any(Project.class));
```

Comment:
Co to za potwór, rodziel wheny rozdziel reesponse rozdziel asserty i rozdziel verify